subprojects/core/src/main/java/org/gradle/execution/plan/OrdinalNodeAccess.java

```diff
@@ -35,25 +35,29 @@ public OrdinalNodeAccess(OrdinalGroupFactory ordinalGroups) {
 
     void addDestroyerNode(OrdinalGroup ordinal, LocalTaskNode destroyer, Consumer<Node> ordinalNodeConsumer) {
         // Create (or get) a destroyer ordinal node that depends on the output locations of this task node
-        ordinal.getDestroyerLocationsNode().addDependenciesFrom(destroyer);
+        OrdinalNode destroyerLocations = ordinal.getDestroyerLocationsNode();
+        destroyerLocations.addDependenciesFrom(destroyer);
+        maybeSchedule(ordinalNodeConsumer, destroyerLocations);
```

Comment:
I think we should avoid scheduling these nodes here, because the node does not have anything that depends on it and they can potentially have a very large number of dependencies. We used to eagerly schedule these nodes but stopped doing so as part of recent performance improvements.

It's not clear to me why we need to eagerly schedule the destroyer locations node here. If the node later happens to be required (ie we discover a producer in a later group) then it will be scheduled in `addProducerNode()` below when the dependency on it is added. The destroy locations nodes for earlier groups will also be added, because `OrdinalGroup.getProducerLocationsNode()` adds the dependency when the node is created.